### ISSUE TITLE
Unable to issue queries with an exclamation mark in them

### ISSUE THREAD

**Issue by MatthewSteeples** [microsoft/AzureStorageExplorer#4599]
### Preflight Checklist

- [X] I have installed the [latest version of Storage Explorer](https://github.com/Microsoft/AzureStorageExplorer/releases/latest).
- [X] I have checked existing resources, including the [troubleshooting guide](https://docs.microsoft.com/azure/storage/common/storage-explorer-troubleshooting) and the [release notes](https://github.com/microsoft/AzureStorageExplorer/releases).
- [X] I have searched for [similar issues](https://github.com/Microsoft/AzureStorageExplorer/issues).

### Storage Explorer Version

1.19.1

### Regression From

Unknown

### Architecture

x64

### Storage Explorer Build Number

20210425.1

### Platform

Windows

### OS Version

Windows 10 19043.1081

### Bug Description

When trying to issue a table query for a property with an exclamation mark in it `!`, the character does not get encoded into a URL friendly format `%21` but instead gets translated into `not`.

### Steps to Reproduce

1. Launch Storage Explorer and navigate to a table
2. Attach Fiddler or another tool to capture network traffic
3. In the filter, type a query string that has an exclamation mark in it eg `123!`
4. Observe in the network traffic that the query string contains the word `not` (instead of `%21`)

### Actual Experience

Top example is what the .NET SDK produces, bottom 3 are Azure Storage Explorer
![image](https://user-images.githubusercontent.com/255971/122764744-b1370600-d297-11eb-94d4-98cb5e87f877.png)

**Comment by JasonYeMSFT** [comment]
Related to https://github.com/Azure/azure-storage-node/issues/588

**Comment by craxal** [comment]
@MatthewSteeples I'm assuming you're attempting to send a query like this?
```
Value eq '123!'
```

We've switched to using @azure/data-tables, and it looks like it's sending the correct query format:
```
GET https://account.table.core.windows.net/mytable()?$select=&$filter=Value%20eq%20%27123!%27 HTTP/1.1
```

You should see this fixed in 1.22.0.

**Comment by MatthewSteeples** [comment]
@craxal Correct, that's the kind of querystring we've been trying. Looking forward to 1.22.0 :)
